packages/fxa-payments-server/src/routes/Product/IapRoadblock/index.tsx

```diff
@@ -39,29 +39,26 @@ export const IapRoadblock = ({
   isMobile,
   selectedPlan,
   subscription,
+  code,
 }: IapRoadblockProps) => {
   const { l10n } = useLocalization();
   const mobileAppStore = l10n.getString(
     getIapSubscriptionAppStoreL10Id(subscription)
   );
-  const appStoreLink = getIapSubscriptionManagementUrl(subscription);
   const subtitle = <p />;
   const title = (
-    <SubscriptionTitle screenType="iapsubscribed" subtitle={subtitle} />
+    <SubscriptionTitle screenType="iaperrorupgrade" subtitle={subtitle} />
   );
 
-  const manageSubscription: VoidFunction = () =>
-    (window.location.href = appStoreLink);
-
   return (
     <>
       <Header {...{ profile }} />
       <div className="main-content">
         <PaymentErrorView
           {...{
             subscriptionTitle: title,
-            error: { code: 'iap_already_subscribed' },
```

Comment:
With this change, are we "removing" the `iap_already_subscsribed` `PaymentErrorView`?

Would it be better to keep the PaymentErrorView for `iap_already_subscribed`, when the user is trying to subscribe to the same plan they already have an IAP subscription for?

And then add a new PaymentErrorView when the user is trying to upgrade.

Reply:
That is still there, but the code was updated to pass in a string dynamically, as opposed to a particular one, for custom error messaging:

`iap_already_subscribed` (l10n id: `iap-already-subscribed`) has now been replaced with `iap_upgrade_already_subscribed` (l10n id: `iap-upgrade-already-subscribed`). Copy difference below:

```
[OLD] iap-already-subscribed = You’re already subscribed through the { $mobileAppStore }.
[NEW] iap-upgrade-already-subscribed = You already have a { $productName } subscription via the { $mobileAppStore }.
```

In addition, two new custom error messages have been added, so if a user encounters either scenario, the correct string gets passed in (l10n ids provided below):

```
iap-upgrade-no-bundle-support = We don’t support bundle upgrades for these subscriptions, but we will soon.
iap-upgrade-contact-support = You can still get the bundle — please contact support so we can help you.
```

Reply:
The `PaymentErrorView` button will be different though right?

Instead of having a button to navigate the user to manage subscription, it'll show the Get Help button, which links them to an "Upgrade Error" page, which isn't really appropriate if the user isn't trying to upgrade.

Reply:
Just adding some more detail.

For example, assume a customer is subscribed to Plan A, via IAP.

In scenario 1, they navigate to the subscription page for Plan A on the web.
-> Expected outcome should be: `PaymentErrorView` with a button that navigates them to Manage their IAP subscription

In scenario 2, they navigate to the subscription page for Plan B, which is an upgrade to Plan A, on the web.
-> Expected outcome should be: `PaymentErrorView` with a button labeled "Get help", which should navigate them to the "Upgrade error" page "https://support.mozilla.org/en-US/kb/upgrade-error"

Reply:
Ah, I see what you are saying. I initially thought the same thing and was going to raise it with design, but then I considered how the error returned from scenario 1 is also the default when none of the other programmed cases match (essentially the same as `INVALID`). As a user, I wouldn't know this and let's say I really was trying to do something completely different and was directed to this page. My instinct would be to contact support/get help. But you are right that in the actual case of scenario 1 (maybe a user forgot about their subscription or went via the web or clicked a link, etc.), then a button linking to my subscription(s) would make more sense.

An alternative would be to maybe add buttons for both? "Get help" above with "Manage my subscription" below?

Reply:
Just taking a step back, considering the example mentioned above, currently the Subscription page handles scenario 1. As far as I know, this ticket is meant to add support for scenario 2.

So I don't think we should change the behaviour for scenario 1, and only add the support for scenario 2. Do you agree?

Reply:
For documentation purposes, after further discussions with Chelsea, I will be making the following updates:

- re-add the old flow
- merge the 3 errors into 1 and provide the new flow with updated button